### PR TITLE
Implement  mkldnn spaceToBatch node dynamic shape support

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:
 - *Implement  mkldnn spaceToBatch node dynamic shape support*
 - *...*

### Tickets:
 - *70995*

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_batch.cpp [hunk 1]
```diff
@@ -2,77 +2,98 @@
...
 using namespace ngraph::opset3;
+using namespace ov::test;
 
 namespace CPULayerTestsDefinitions  {
 
-typedef std::tuple<
-        LayerTestsDefinitions::spaceToBatchParamsTuple,
-        CPUSpecificParams> SpaceToBatchLayerCPUTestParamSet;
+using SpaceToBatchLayerTestCPUParams = std::tuple<
+        std::vector<InputShape>,            // Input shapes
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_batch.cpp:17]: as I can see in this test we have dynamic shapes only on `data`
so looks like enough `InputShape` instead vector

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_batch.cpp [hunk 2]
```diff
@@ -2,77 +2,98 @@
...
-                ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
-        auto s2b = ngraph::builder::makeSpaceToBatch(paramOuts[0], ngPrc, blockShape, padsBegin, padsEnd);
-        s2b->get_rt_info() = getCPUInfo();
-        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(s2b)};
+        auto params = ngraph::builder::makeDynamicParams(ngPrec, {inputDynamicShapes.front()});
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+        auto b2s = ngraph::builder::makeSpaceToBatch(paramOuts[0], ngPrec, blockShape, padsBegin, padsEnd);
+        b2s->get_rt_info() = getCPUInfo();
+        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(b2s)};
         function = std::make_shared<ngraph::Function>(results, params, "SpaceToBatch");
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_batch.cpp:82]: the same comments: https://github.com/openvinotoolkit/openvino/pull/8639#discussion_r753651630

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_batch.cpp [hunk 3]
```diff
@@ -2,77 +2,98 @@
...
     }
 };
 
 TEST_P(SpaceToBatchCPULayerTest, CompareWithRefs) {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
 
-    Run();
-    CheckPluginRelatedResults(executableNetwork, "SpaceToBatch");
+    run();
+    // TODO: Should be uncommented after updating the CheckPluginRelatedResults() method
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_batch.cpp:90]: uncomment please

**Comment by luweizhou2016** [comment]
Applied review comments.

**Review by mandrono — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**